### PR TITLE
Add test for func in cobra.go

### PR DESCRIPTION
**Pull request by korovindenis**
Add test for:

- rpad(s string, padding int) string
- ld(s, t string, ignoreCase bool) int
- stringInSlice(a string, list []string) bool

in cobra.go

### CODE REVIEW

File: cobra_test.go [hunk 1]
```diff
@@ -40,3 +40,185 @@ func TestAddTemplateFunctions(t *testing.T) {
...
+		},
+		{
+			name:       "One empty string",
+			s:          "abc",
+			t:          "",
+			ignoreCase: false,
+			expected:   3,
+		},
+		{
+			name:       "Both empty strings",
```
Comment (marckhouzam) [inline cobra_test.go:95]: Is this kind of a duplicate of the test called `Empty strings`, or does the `ignoreCase` matter?

File: cobra_test.go [hunk 2]
```diff
@@ -40,3 +40,185 @@ func TestAddTemplateFunctions(t *testing.T) {
...
+			expected: true,
+		},
+		{
+			name:     "String not in slice (case-sensitive)",
+			a:        "pear",
+			list:     []string{"orange", "banana", "apple", "grape"},
+			expected: false,
+		},
+		{
+			name:     "String in slice (case-insensitive)",
```
Comment (marckhouzam) [inline cobra_test.go:136]: I don't think we need case-insensitive tests since the function being testes is case-sensitive.
So, for this test, since `apple` does not match in the test above, do we really need to test `APPLE`?

File: cobra_test.go [hunk 3]
```diff
@@ -40,3 +40,185 @@ func TestAddTemplateFunctions(t *testing.T) {
...
+			a:        "",
+			list:     []string{"orange", "banana", "apple", "grape"},
+			expected: false,
+		},
+		{
+			name:     "Empty string in empty slice",
+			a:        "",
+			list:     []string{},
+			expected: false,
+		},
```
Comment (marckhouzam) [inline cobra_test.go:164]: How about adding a test that empty strings can match?
```
		{
			name:     "Empty strings match",
			a:        "",
			list:     []string{"orange", ""},
			expected: true,
		},
```

**Review by marckhouzam — COMMENTED** [review]
Thanks @korovindenis.
More tests is always better.
Some small nits if you agree with them.

File: cobra_test.go (hunk 1 above)
Comment (korovindenis) [reply cobra_test.go:95]: These tests are similar, but ignoreCase matters.

File: cobra_test.go (hunk 2 above)
Comment (korovindenis) [reply cobra_test.go]: Agreed, this test is unnecessary

File: cobra_test.go (hunk 3 above)
Comment (korovindenis) [reply cobra_test.go:164]: Yes, that's a good scenario. Added the test.

File: cobra_test.go (hunk 2 above)
Comment (marckhouzam) [reply cobra_test.go]: Sorry I got mixed up. `apple` matches in the above test so checking that `APPLE` does not match has value. It confirms the function is case-sensitive. 

It is the `PEAR` test that is not needed

**Review by marckhouzam — APPROVED** [review]
Thank you!
